feature-configs/cn-ran-overlays/hacks/pre_deploy.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/bash
+
+set -e
+
+absolute_feature_dir="$(pwd)/$(dirname $0)"
+feature=$(basename ${absolute_feature_dir})
+cluster_type="${CLUSTER_TYPE:-standard}"
+ztp_generator_cli=ztp/resource-generator/entrypoints/generator
+
+du_source_profile="ran-profile"
+du_ci_standard_profile="ran-profile-gcp"
+du_features=($du_source_profile $du_ci_standard_profile)
+
+if [ $feature == $du_ci_standard_profile ]; then
+    cluster_type="standard"
+fi
+
+# set variables for ztp generator cli
+export policyGenPath=${absolute_feature_dir}/policyGen
+
+# prepare resources required by ztp generator cli
+mkdir -p ${policyGenPath}
+cp -r ztp/source-crs ${policyGenPath}
+go build -o ${policyGenPath}/PolicyGenTemplate ./ztp/policygenerator
+
+if [[ "${du_features[@]}" =~ "$feature" ]]; then
+    source_pgt_dir=$(dirname ${absolute_feature_dir})/${du_source_profile}/policygentemplates
+
+    source_pgts=("common-ranGen.yaml")
+    if [ $cluster_type == "standard" ];then
+        source_pgts+=("group-du-standard-ranGen.yaml" "example-multinode-site.yaml")
+    elif [ $cluster_type == "sno" ]; then
+        source_pgts+=("group-du-sno-ranGen.yaml" "example-sno-site.yaml")
+    elif [ $cluster_type == "3node" ]; then
+        source_pgts+=("group-du-3node-ranGen.yaml" "example-multinode-site.yaml")
+    fi
+
+    # call ztp generator cli to generate the non-policy wrapped cluster config CRs based on the source PGTs
+    for source_pgt in "${source_pgts[@]}"; do
+        source_config_CRs="${ztp_generator_cli} config -N ${source_pgt_dir}/${source_pgt} ${absolute_feature_dir}/cluster-config"
+        if ! $source_config_CRs; then
+            echo "[ERROR] Failed to generate the configuration CRs based on the source PGT: ${source_pgt_dir}/${source_pgt}"
+            exit 1
+        fi
+    done
+
+    # call ztp generator cli to generate the configuration CRs for CI environment
+    if [ $feature != $du_source_profile ]; then
+        ci_config_CRs="${ztp_generator_cli} config -N ${absolute_feature_dir}/policygentemplates ${absolute_feature_dir}/cluster-config"
+        if ! $ci_config_CRs; then
+            echo "[ERROR] Failed to generate the configuration CRs based on the overrided PGTs for CI environment: ${absolute_feature_dir}/policygentemplates"
+            exit 1
+        fi
+    fi
+fi
```

Comment:
If I understand correctly this block is generating configuration CRs that are specific to the CI environment and the block above (lines 39-45) are generating configuration CRs from the reference. It is a bit confusing that both contain "common-rangen.yaml" and "group-du-standard-ranGen.yaml". Is the intent that this later set is overwriting the reference versions? Or are they overwriting just parts of the reference versions? A note to clarify how these two blocks relate would be helpful.

Reply:
The PGTs in the ran-profile-gcp repo override the parts of the reference versions that are for CI env.  Will update.